tests/test_unified_install_cache.py

```diff
@@ -0,0 +1,208 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import json
+import os
+import subprocess
+from subprocess import CalledProcessError
+from textwrap import dedent
+
+import pytest
+
+from pex import pip
+from pex.common import safe_mkdir, safe_mkdtemp, safe_open, safe_rmtree
+from pex.pex_info import PexInfo
+from pex.testing import run_pex_command
+
+
+@pytest.fixture(scope='module')
+def pex_project_dir():
+  return subprocess.check_output(['git', 'rev-parse', '--show-toplevel']).decode('utf-8').strip()
+
+
+def test_issues_789_demo(pex_project_dir):
```

Comment:
I really appreciate the way the pex codebase maps github issues to resolve to specific tests and comments in the code!